src/main/msp/msp.c

```diff
@@ -1843,6 +1859,35 @@ static mspResult_e mspFcProcessOutCommandWithArg(uint8_t cmdMSP, sbuf_t *src, sb
         break;
 #endif // USE_VTX_TABLE
 
+    case MSP_RESET_CONF:
+        {
+#if defined(USE_CUSTOM_DEFAULTS)
+            defaultsType_e defaultsType = DEFAULTS_TYPE_CUSTOM;
+#endif
+            if (sbufBytesRemaining(src) >= 1) {
+                // Added in MSP API 1.42
+#if defined(USE_CUSTOM_DEFAULTS)
+                defaultsType = sbufReadU8(src);
+#else
+                sbufReadU8(src);
+#endif
+            }
+
+            if (!ARMING_FLAG(ARMED)) {
```

Comment:
should this return an error response if you try and reset defaults when armed?  otherwise the clients will think they succeeded.

Reply:
Potentially. The existing implementation doesn't return anything, so clients won't be checking for a response.
Also, this pull request introduces a reboot in the case that the reset is successful, so clients most likely won't have time to do anything meaningful with the response before the flight controller disconnects.

Reply:
if the FC doesn't reboot the client might get stuck, waiting for a reboot.  i think an error code here would be best so that client authors can handle error cases without having to timeout waiting for a reboot.